Add dependabot (non-preview version)

marco-neumann-by opened the pull request:
Now that the dependabot preview was disabled on org-level, let's bring
back the GitHub-builtin version.

Background: The original dependabot company was bought by GitHub and is
now part of GitHubs services. See
https://docs.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates

gabriel-kohen-by reviewed (changes requested):
You might find `daily` to be bothersome. It will still do security updates daily as they come. The schedule it is for non-security related updates. Maybe weekly is better.

marco-neumann-by committed:
Add dependabot (non-preview version)

Now that the dependabot preview was disabled on org-level, let's bring
back the GitHub-builtin version.

Background: The original dependabot company was bought by GitHub and is
now part of GitHubs services. See
https://docs.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates

gkohen commented:
@marco-neumann-by is this still needed?
